.github/workflows/ci.yaml

```diff
@@ -0,0 +1,52 @@
+name: Continuous Integration
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+
+jobs:
+  runner:
+    runs-on: ubuntu-latest
+    # We use the same job template, but parametrize the actual command to be passed to the runner
+    # binary using the matrix strategy, so that we get the commands running in parallel.
+    strategy:
+      fail-fast: false
+      matrix:
+        cmd:
+          - check-format
+          - run-cargo-deny
+          - run-cargo-udeps
+          - build-server --server-variant=base
+          - build-server --server-variant=logless
+          - run-tests
+          - run-tests-tsan
+          - run-examples --application-variant=rust
+          - run-examples --application-variant=cpp
```

Comment:
The way that GitHub displays individual steps makes it hard to see which step is which among these &ndash; I guess there's no way around that?

Reply:
Not that I can think of, anyways it seems ok to me, these do show up in parentheses next to the step name.

Reply:
![image](https://user-images.githubusercontent.com/457208/100007391-e93e9f00-2dc3-11eb-9775-23e3817f26f9.png)

Reply:
Uhmm god it. Anyways it may be possible to specify objects with {name, command} fields in the matrix and extract the two individually, I may try that separately but also not sure it's worth it.